mllib/src/test/scala/org/apache/spark/ml/util/DefaultReadWriteTest.scala

```diff
@@ -64,7 +64,12 @@ trait DefaultReadWriteTest extends TempDirectory { self: Suite =>
             case (Array(values), Array(newValues)) =>
               assert(values === newValues, s"Values do not match on param ${p.name}.")
             case (value, newValue) =>
-              assert(value === newValue, s"Values do not match on param ${p.name}.")
+              if (value.isInstanceOf[Double] && value.asInstanceOf[Double].isNaN) {
```

Comment:
Why we need this check?

If necessary, may change to:
```
case (value: Double, newValue: Double) =>
      assert(value.isNaN && newValue.isNaN || value == newValue, ...)
```

Reply:
Before my change, this ```setDefault(strategy -> Imputer.mean, missingValue -> Double.NaN)``` is in Imputer. Now I moved this to ImputerParams so ImputerModel also has the default value of ```missingValue``` set to ```Double.NaN```. In ```DefaultReadWriteTest```, it compares the default value of ```missingValue``` in the original model and the loaded model, so I need to make the NaN comparison work.